From bee2a942be50c7a879e74a1c0a76c927cae9048f Mon Sep 17 00:00:00 2001
From: Cyan <cyan@cyano.uk>
Date: Fri, 22 Dec 2023 21:17:03 +0800
Subject: [PATCH] configure.ac: fix -fzero-call-used-regs=used for mips64
 targets

- This check somehow succeeded during configure check, but failed during
  build.
---
 configure.ac | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 379cd74..e36cc60 100644
--- a/configure.ac
+++ b/configure.ac
@@ -236,7 +236,16 @@ if test "$GCC" = "yes" || test "$GCC" = "egcs"; then
 	case "$CLANG_VER" in
 	apple-15*) OSSH_CHECK_CFLAG_LINK([-fzero-call-used-regs=used]) ;;
 	17*)	;;
-	*)	OSSH_CHECK_CFLAG_LINK([-fzero-call-used-regs=used]) ;;
+	*)
+		case "$host" in
+			# FIXME possible GCC oversight - 
+			# sorry, unimplemented: argument ‘used’ is not supported for ‘-fzero-call-used-regs’ on this target
+			# This check passed during configure, but failed later during build.
+			# It should fail during this check.
+			mips64el*) 	;;
+			*)		OSSH_CHECK_CFLAG_LINK([-fzero-call-used-regs=used]) ;;
+		esac
+		;;
 	esac
 	OSSH_CHECK_CFLAG_COMPILE([-ftrivial-auto-var-init=zero])
     fi
-- 
2.39.1

